src/components/page/page_header/page_header.test.tsx

```diff
@@ -124,4 +124,48 @@ describe('OuiPageHeader', () => {
       });
     });
   });
+
+  describe('deprecation', () => {
+    it('should console 1 deprecation warning without repetition', () => {
+      console.warn = jest.fn();
+
+      const component = mount(<OuiPageHeader iconType="dashboardApp" />);
+      component.setProps({ iconType: 'database' });
+
+      expect(console.warn).toHaveBeenCalledTimes(1);
+      expect(console.warn).toHaveBeenCalledWith(
+        '[DEPRECATED] The `iconType` prop is deprecated and will be removed in v2.0.0'
+      );
+    });
+
+    it('should console 2 deprecation warning without repetition', () => {
+      console.warn = jest.fn();
+
+      const component = mount(
+        <OuiPageHeader iconType="dashboardApp" iconProps={{ color: 'red' }} />
+      );
+      component.setProps({
+        iconType: 'database',
+        iconProps: { color: 'blue' },
+      });
+
+      const results = [
+        '[DEPRECATED] The `iconType` prop is deprecated and will be removed in v2.0.0',
+        '[DEPRECATED] The `iconProps` prop is deprecated and will be removed in v2.0.0',
+      ];
+
+      expect(console.warn).toHaveBeenCalledTimes(2);
+      results.forEach((item) =>
+        expect(console.warn).toHaveBeenCalledWith(item)
+      );
+    });
+
+    it('should not console deprecation warning', () => {
+      console.warn = jest.fn();
+
+      mount(<OuiPageHeader />);
```

Comment:
It's possible this _could_ be deprecated sometime later down the line, but I think it's fine for now. We can worry about that later :)